examples/mpi_shallow_water_turbulence.jl

```diff
@@ -0,0 +1,78 @@
+using MPI
+
+MPI.Initialized() || MPI.Init()
+
+using Statistics
+using Oceananigans
+using Oceananigans.Distributed
+
+topo = (Periodic, Periodic, Bounded)
+full_grid = RegularRectilinearGrid(topology=topo, size=(128, 128, 1), extent=(4π, 4π, 1), halo=(3, 3, 3))
+arch = MultiCPU(grid=full_grid, ranks=(1, 4, 1))
+
+model = DistributedShallowWaterModel(
+                  architecture = arch,
+                          grid = full_grid,
+                   timestepper = :RungeKutta3,
+                     advection = UpwindBiasedFifthOrder(),
+                       closure = IsotropicDiffusivity(ν=1e-5),
+    gravitational_acceleration = 1.0
+)
+
+set!(model, h=model.grid.Lz)
+
+uh₀ = rand(size(model.grid)...);
+uh₀ .-= mean(uh₀);
+set!(model, uh=uh₀, vh=uh₀)
+
+progress(sim) = @info "Iteration: $(sim.model.clock.iteration), time: $(sim.model.clock.time)"
+simulation = Simulation(model, Δt=0.001, stop_time=2, iteration_interval=1, progress=progress)
+
+uh, vh, h = model.solution
+outputs = (ζ=ComputedField(∂x(vh/h) - ∂y(uh/h)),)
+filepath = "mpi_shallow_water_turbulence_rank$(arch.my_rank).nc"
+simulation.output_writers[:fields] =
+    NetCDFOutputWriter(model, outputs, filepath=filepath, schedule=TimeInterval(0.01), mode="c")
+
+MPI.Barrier(MPI.COMM_WORLD)
```

Comment:
Life goes better if at the very start (just after MPI_init) you create a local MPI Communicator name e.g. 
```
 MPI_Comm_model = MPI.COMM_WORLD
```
and then use that.  That way when we come to couple, run ensembles under MPI etc.... it will be easier to
introduce a local communicator.

Reply:
Ah I didn't appreciate this, would be a good change to make!

Would it make sense to store the `MPI.COMM_WORLD` after `MPI.Init()` as part of the multi-architecture?

E.g. `MPI.COMM_WORLD` is passed to the `MultiCPU` constructor and then stored under `model.architecture.communicator`? This way it should be pretty easy to access the communicator almost anywhere in the code.